bin/lib/vets-api/commands/setup.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require_relative '../setups/base'
+require_relative '../setups/native'
+require_relative '../setups/docker'
+require_relative '../setups/hybrid'
+require './rakelib/support/shell_command'
+
+module VetsApi
+  module Commands
+    class Setup
+      class << self
+        # TODO: run check to make sure in the correct directory /vets-api
+        def run(args)
+          setup_base
+          unless args.first == '--base'
+            store_developer_environment_preference(args.first)
+            setup_developer_environment
+          end
+        end
+
+        private
+
+        def setup_base
+          VetsApi::Setups::Base.new.run
+        end
+
+        def store_developer_environment_preference(input_environment)
+          file_path = '.developer-setup'
+          File.write(file_path, input_environment.tr('--', '')) if input_environment
+        end
+
+        def setup_developer_environment
+          case File.read('.developer-setup').chomp
+          when 'native'
+            setup_native
+          when 'docker'
+            setup_docker
+          when 'hybrid'
+            setup_hybrid
+          else
+            puts 'Invalid option for .developer-setup'
+          end
+        end
+
+        def setup_native
+          if RUBY_DESCRIPTION.include?('ruby 3.2.4')
```

Comment:
Can we extract this version from the .ruby-version file so we don't have a lot of places to change the version when there's an update?